addons/account/tests/test_account_analytic.py

```diff
@@ -402,3 +402,73 @@ def test_analytic_applicability_multiple_prefixes(self):
 
         # This invoice should not be blocked, as all lines have plans
         invoice.with_context({'validate_analytic': True}).action_post()
+
+    def test_synchronization_between_analytic_distribution_and_analytic_lines(self):
+        """ Test creating, updating, and deleting analytic lines and ensure the changes are reflected in move_line's analytic_distribution. """
+        
+        # Create analytic accounts for distribution
+        analytic_account_1 = self.env['account.analytic.account'].create({
+            'name': 'Analytic Account 1',
+            'plan_id': self.cross_plan.id,
+        })
+        analytic_account_2 = self.env['account.analytic.account'].create({
+            'name': 'Analytic Account 2',
+            'plan_id': self.cross_plan.id,
+        })
+        
+        # Create a product and a partner for the invoice
+        product = self.env['product.product'].create({
+            'name': 'Test Product',
+            'type': 'service',
+            'list_price': 100.0,
+        })
+        partner = self.env['res.partner'].create({
+            'name': 'Test Partner',
+        })
+        
+        # Create an invoice with analytic distribution
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': partner.id,
+            'date': '2023-01-01',
+            'invoice_date': '2023-01-01',
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': product.id,
+                    'price_unit': 100.0,
+                    'analytic_distribution': {
+                        str(analytic_account_1.id): 40,
+                        str(analytic_account_2.id): 60,
+                    },
+                })
+            ]
+        })
+        
+        # Post the invoice
+        invoice.action_post()
+
+        # Fetch the associated move line and analytic lines
+        analytic_lines = self.env['account.analytic.line'].search([
+                ('move_line_id', 'in', invoice.line_ids.ids)
+            ]).sorted('amount')
+        move_line = analytic_lines[0].move_line_id
+        
+        # Update the account of the first analytic line
+        analytic_lines[0].account_id = analytic_account_2
+        analytic_lines[0].amount = 50  # Adjust amount after account change
```

Comment:
Also test writing on `move_line_id`.
It should update the distribution of both the old line (before the write) and the new line (after the write)